dev/source-release.sh

```diff
@@ -20,72 +20,137 @@
 
 set -e
 
-if [ -z "$1" ]; then
-  echo "Usage: $0 <version-num> <rc-num>"
-  exit
-fi
-
-if [ -z "$2" ]; then
-  echo "Usage: $0 <version-num> <rc-num>"
-  exit
+usage () {
+    echo "usage: $0 [-k <key_id>] [-r <git_remote>] <version_num> <rc_num>"
+    echo "  -k      Specify signing key. Defaults to \"GPG default key\""
+    echo "  -r      Specify Git remote name. Defaults to \"origin\""
+    echo "  -d      Turn on DEBUG output"
+    exit 1
+}
+
+# Default repository remote name
+remote="origin"
+
+while getopts "k:r:d" opt; do
+  case "${opt}" in
+    k)
+      keyid="${OPTARG}"
+      ;;
+    r)
+      remote="${OPTARG}"
+      ;;
+    d)
+      set -x
+      ;;
+    *)
+      usage
+      ;;
+  esac
+done
+
+shift $((OPTIND-1))
+
+version="$1"
+rc="$2"
+
+if [ -z "$version" ] || [ -z "$rc" ]; then
+  usage
```

Comment:
Non-blocking / open-ended question:

Can / should we add a block comment at the top of the script, specifying that this is for release candidates only?

The script name was a bit confusing for me for a while, as `rc` is required and it seems this is only for pushing release candidates. Some small indicator might be a good idea. 

I often see shell script top level comments like this, though I don't have a huge preference. My follow up question would be possibly documenting `version` and `rc`, though those are arguably self-documenting from the usage function (I'd still love to see them as `getopts` but I'm open to discussion on that / doing that later).

```bash
#
# Generate a release-candidate and a text file containing the release announcement
#
```

Reply:
Done